Xieting Zhang, HW2, Biostat 578 Bioinformatics
========================================================

The following codes and annotations include all steps for completing homework 2.

First of all, we need to install bioconductor packages and install the GEOmetadb package which contains the data
we will use. 
```{r}
source("http://bioconductor.org/biocLite.R")
biocLite()
biocLite(c("GEOmetadb", "GEOquery"))
```

Before we can use the GEOmetadb package, we need to call the library
first. 
```{r }
library(GEOmetadb)
```

We also need to make a connection with the database, install limma package and call all the 
libraries we will use. 

```{r }
geo_con <- dbConnect(SQLite(),'GEOmetadb.sqlite')
biocLite("limma")
library(limma)
library(Biobase)
library(data.table)
```


Now we can get the data first:

```{r }
mygds <- getGEO("GSE40812")
```

Since we have downloaded the processed data rather than the raw data, we do not need to normalize it. But we can
clean it up somehow:

```{r }
mygds_new <- mygds
sanitize_pdata <- function(pd){
  keepCols <- c(
    "characteristics_ch1", "characteristics_ch1.2")
  pd <- pd[, keepCols]
  colnames(pd) <- c("pt", "treatment")
  pd$pt <- gsub(".*: ", "", pd$pt)
  pd$treatment<-gsub("treatment: ", "", pd$treatment)
  pd
}
pData(mygds_new[[1]]) <- sanitize_pdata(pData(mygds_new[[1]]))
```

To use the limma package and its regression function, we need to have the model matrix set up 
the design matrix:

```{r }
mm <- model.matrix(~pt+treatment, mygds_new[[1]])

```

Then we can run the linear fit and use topTable function to find the genes that
are differentially expressed:

```{r }
mfit<- lmFit(mygds_new[[1]], mm)
ebay_m <- eBayes(mfit)
topp <- topTable(ebay_m, coef="treatmentPoly IC H", adjust='fdr', p.value = 0.05)
```

We need to pick out the genes that have an adjusted p-value less than 0.1:
```{r }
selected <- p.adjust(ebay_m$p.value[,3]) < 0.1
gds.selected <- mygds_new[[1]][selected,]
```

The we can plot the heatmap:
```{r}
install.packages("pheatmap")
library(pheatmap)
heatmap(exprs(gds.selected))
```
